### PR TITLE
test(unify): Reporter header e2e tests

### PR DESCRIPTION
**Pull request by davidmunechika**
- Closes https://github.com/cypress-io/cypress/issues/19218

### User facing changelog
n/a

### Additional details

Added the following additional e2e tests to finish the Reporter Header sections on the spreadsheet:
- The currently running spec will display as highlighted in the Specs List (update: removed from E2E spreadsheet section and updated CT test)
- The extension in the list will display as a different highlighted color than the filename (update: removed from E2E spreadsheet section since better suited for CT test)
- Searching the specs will filter the list of specs
- Clicking the down arrow will open a panel showing Testing Preferences
- The auto-scrolling will show a toggle beside it.

### PR Tasks

- [x] Have tests been added/updated?
- [n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [n/a] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/app/cypress/e2e/reporter_header.cy.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+      cy.get('[data-testid=spec-file-item]').get('.text-gray-700')
+      .should('contain', '.')
+      .should('contain', 's')
+      .should('contain', 'p')
+      .should('contain', 'e')
+      .should('contain', 'c')
+      .should('contain', '.')
+      .should('contain', 'j')
+      .should('contain', 's')
```
[the remark this answers is not in the document]
Comment (davidmunechika) [reply packages/app/cypress/e2e/reporter_header.cy.ts:33]: i thought i tried that lol...let me try again 😅

[the remark this answers is not in the document]
Comment (davidmunechika) [reply packages/app/cypress/e2e/reporter_header.cy.ts:33]: Screenshot 2022-02-03 at 2 18 37 PM

yeah `.should('eq', '.spec.js')` doesn't pass... 🤔

[the remark this answers is not in the document]
Comment (flotwig) [reply packages/app/cypress/e2e/reporter_header.cy.ts:33]: seems like there are multiple matched elements, maybe that's why?

File: packages/app/cypress/e2e/reporter_header.cy.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    cy.contains('dom-content.spec').click()
+  })
+
+  context('Specs Shortcut', () => {
+    it('displays the currently running spec as highlighted in the Specs List', () => {
+      cy.location().should((location) => {
+        expect(location.hash).to.contain('dom-content.spec')
+      })
+
+      cy.get('[data-selected-spec="true"').should('contain', 'dom-content')
```
Comment (emilyrohrbough) [inline packages/app/cypress/e2e/reporter_header.cy.ts:16]: Should we be assert this length is 1?

File: packages/app/cypress/e2e/reporter_header.cy.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      cy.get('[data-selected-spec="false"').should('have.length', '1')
```
Comment (emilyrohrbough) [inline packages/app/cypress/e2e/reporter_header.cy.ts:17]:
```suggestion
      cy.get('[data-selected-spec="false"]').should('have.length', '1')
```

File: packages/app/cypress/e2e/reporter_header.cy.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  beforeEach(() => {
+    cy.scaffoldProject('cypress-in-cypress')
+    cy.openProject('cypress-in-cypress')
+    cy.startAppServer()
+    cy.visitApp()
+    cy.contains('dom-content.spec').click()
+  })
+
+  context('Specs Shortcut', () => {
+    it('displays the currently running spec as highlighted in the Specs List', () => {
```
Comment (emilyrohrbough) [inline packages/app/cypress/e2e/reporter_header.cy.ts:11]: This doesn't appear to to be verify the spec is highlighted, just has the selected data attribute. Maybe the name is mis-leading? Or are there additional check missing?

File: packages/app/cypress/e2e/reporter_header.cy.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      cy.location().should((location) => {
+        expect(location.hash).to.contain('dom-content.spec')
+      })
+
+      cy.get('[data-selected-spec="true"').should('contain', 'dom-content')
+      cy.get('[data-selected-spec="false"').should('have.length', '1')
+    })
+
+    it('will display the extension in the list as a different highlighted color than the filename', () => {
```
Comment (emilyrohrbough) [inline packages/app/cypress/e2e/reporter_header.cy.ts:20]: This test seems to suggest the filename + extesion are one color, not different.

File: packages/app/cypress/e2e/reporter_header.cy.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      cy.location().should((location) => {
+        expect(location.hash).to.contain('dom-content.spec')
+      })
+
+      cy.get('input').type('dom', { force: true })
+
+      cy.get('[data-testid="spec-file-item"]').should('have.length', 1)
+      .should('contain', 'dom-content.spec')
+
+      cy.get('input').clear().type('asdf', { force: true })
```
Comment (emilyrohrbough) [inline packages/app/cypress/e2e/reporter_header.cy.ts:46]: can we validate the clear functionally worked correctly?

```suggestion
      cy.get('input').clear()
      
      cy.get('[data-testid="spec-file-item"]').should('have.length', ) // how many?

      cy.get('input').type('asdf', { force: true })
```

File: packages/app/cypress/e2e/reporter_header.cy.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+      cy.location().should((location) => {
+        expect(location.hash).to.contain('dom-content.spec')
+      })
+
+      cy.get('.testing-preferences-toggle').trigger('mouseover')
+      cy.get('.cy-tooltip').should('have.text', 'Open Testing Preferences')
+
+      cy.get('.testing-preferences').should('not.exist')
+      cy.get('.testing-preferences-toggle').should('not.have.class', 'open')
+      cy.get('.testing-preferences-toggle').click()
```
Comment (emilyrohrbough) [inline packages/app/cypress/e2e/reporter_header.cy.ts:63]:
```suggestion
      cy.get('.testing-preferences-toggle').should('not.have.class', 'open').click()
```

File: packages/app/cypress/e2e/reporter_header.cy.ts [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+      cy.get('.testing-preferences-toggle').should('have.class', 'open')
+      cy.get('.testing-preferences').should('be.visible')
+      cy.get('.testing-preferences-toggle').click()
+      cy.get('.testing-preferences-toggle').should('not.have.class', 'open')
+      cy.get('.testing-preferences').should('not.exist')
+    })
+
```
Comment (emilyrohrbough) [inline packages/app/cypress/e2e/reporter_header.cy.ts:70]: Do we need add any tests to validate that these test preference persist for a user between projects and/or open/closing a project?

File: packages/app/cypress/e2e/reporter_header.cy.ts (hunk 6 above)
Comment (davidmunechika) [reply packages/app/cypress/e2e/reporter_header.cy.ts:46]: Screenshot 2022-02-03 at 5 52 37 PM

I used `force: true` since that's what was done in the component tests for the reporter. Looks like removing it throws an error 😕

File: packages/app/cypress/e2e/reporter_header.cy.ts (hunk 8 above)
Comment (davidmunechika) [reply packages/app/cypress/e2e/reporter_header.cy.ts:65]: I'm not sure about this- i'll ask and get an answer 👍

File: packages/app/cypress/e2e/reporter_header.cy.ts (hunk 6 above)
Comment (emilyrohrbough) [reply packages/app/cypress/e2e/reporter_header.cy.ts]: Hmmm in my experience if the element isn't clickable because it's covered by another element, it's likely a problem with the clickabling for the end-user. Is this input actually interactable at this point in the workflow when manually verifying?

File: packages/app/cypress/e2e/reporter_header.cy.ts (hunk 4 above)
Comment (davidmunechika) [reply packages/app/cypress/e2e/reporter_header.cy.ts]: I updated the CT test for this and removed it from the E2E section of the spreadsheet

File: packages/app/cypress/e2e/reporter_header.cy.ts (hunk 5 above)
Comment (davidmunechika) [reply packages/app/cypress/e2e/reporter_header.cy.ts]: I also removed this from the E2E section of the spreadsheet since it is better covered in CT test (percy snapshot)

File: packages/app/cypress/e2e/reporter_header.cy.ts (hunk 6 above)
Comment (davidmunechika) [reply packages/app/cypress/e2e/reporter_header.cy.ts]: That's what's strange- the input interacts perfect fine when I try and type manually at this point in the test. doesn't seem to be covered by anything

File: packages/app/cypress/e2e/reporter_header.cy.ts [hunk 9]
```diff
@@ -8,23 +8,6 @@ describe('Reporter Header', () => {
   })
 
   context('Specs Shortcut', () => {
-    it('displays the currently running spec as highlighted in the Specs List', () => {
```
Comment (emilyrohrbough) [inline packages/app/cypress/e2e/reporter_header.cy.ts]: I will say, validating the correct spec is selected when running is still a valid e2e test case.

Comment (davidmunechika) [reply packages/app/cypress/e2e/reporter_header.cy.ts]: added it back in 👍

**Review by emilyrohrbough — APPROVED** [review]

**Review by flotwig — APPROVED** [review]
